rust/azure_iot_operations_services/src/azure_device_registry.rs

```diff
@@ -0,0 +1,484 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+//! Types for Azure Device Registry operations.
+use adr_name_gen::adr_base_service::client::{
+    AssetEndpointProfileStatus, AssetStatus, CreateDetectedAssetRequestPayload,
+    DatasetsSchemaSchemaElementSchema, DetectedAsset, DetectedAssetDataPointSchemaElementSchema,
+    DetectedAssetDatasetSchemaElementSchema, DetectedAssetEventSchemaElementSchema, Error,
+    EventsSchemaSchemaElementSchema, MessageSchemaReference, Topic,
+    UpdateAssetEndpointProfileStatusRequestPayload, UpdateAssetStatusRequestPayload,
+    UpdateAssetStatusRequestSchema,
+};
+use adr_type_gen::aep_type_service::client::{
+    CreateDiscoveredAssetEndpointProfileRequestPayload, DiscoveredAssetEndpointProfile,
+    SupportedAuthenticationMethodsSchemaElementSchema,
+};
+use azure_iot_operations_protocol::common::aio_protocol_error::AIOProtocolError;
+use core::fmt::Debug;
+use thiserror::Error;
+
+/// Azure Device Registry generated code
+mod adr_name_gen;
+mod adr_type_gen;
+/// Azure Device Registry Client implementation wrapper
+mod client;
+
+pub use client::Client;
+
+// // OPEN QUESTIONS
+// TODO Except connector this client is used somewhere else ?
+// TODO Do we need to validate any required fields ? If so most probbaly need a builder
+
+/// Represents an error that occurred in the Azure Device Registry Client implementation.
+#[derive(Debug, Error)]
+#[error(transparent)]
+pub struct AzureDeviceRegistryError(#[from] ErrorKind);
```

Comment:
```suggestion
pub struct Error(#[from] ErrorKind);
```

To match the other service clients.